services/src/main/scala/cromwell/services/metrics/bard/impl/BardEventingActor.scala

```diff
@@ -0,0 +1,43 @@
+package cromwell.services.metrics.bard.impl
+
+import akka.actor.{Actor, ActorRef, ActorSystem, Props}
+import com.typesafe.config.Config
+import com.typesafe.scalalogging.LazyLogging
+import common.util.StringUtil.EnhancedToStringable
+import cromwell.core.Dispatcher.ServiceDispatcher
+import cromwell.services.metrics.bard.BardEventing.BardEventRequest
+import cromwell.services.metrics.bard.{BardConfig, BardService}
+import cromwell.util.GracefulShutdownHelper.ShutdownCommand
+
+import scala.concurrent.ExecutionContext
+
+class BardEventingActor(serviceConfig: Config, globalConfig: Config, serviceRegistryActor: ActorRef)
+    extends Actor
+    with LazyLogging {
+
+  implicit val system: ActorSystem = context.system
+  implicit val ec: ExecutionContext = context.dispatcher
+
+  lazy val bardConfig: BardConfig = BardConfig(serviceConfig)
+  lazy val bardService: BardService = new BardService(bardConfig.baseUrl, bardConfig.connectionPoolSize)
+
+  override def receive: Receive = {
+    case BardEventRequest(event) if bardConfig.enabled =>
+      bardService.sendEvent(event)
```

Comment:
Warning! A blocking action in a `receive` method means the actor's mailbox can overflow. Are we sure that the bard sending can keep up with the bard request generation?

Incidentally, it also means that all requests are processed in series. Which is fine, but why do we need a `connection-pool-size` option?

Reply:
Its a standard tuning option. The http connection pool means that connections can be re-used for multiple requests, improving performance.